src/pretalx/agenda/management/commands/export_schedule_html.py

```diff
@@ -1,74 +1,230 @@
-from pathlib import Path
+import contextlib
+import logging
+import os
+import re
+import shutil
+import urllib.parse
 from shutil import make_archive
 
-from bakery.management.commands.build import Command as BakeryBuildCommand
+from bs4 import BeautifulSoup
 from django.conf import settings
-from django.core.management.base import CommandError
-from django.test import override_settings
-from django.urls import get_callable
-from django.utils import translation
+from django.core.management.base import BaseCommand, CommandError
+from django.test import Client, override_settings
 from django.utils.timezone import override as override_timezone
 
-from pretalx.event.models import Event
+from pretalx.common.utils import rolledback_transaction
+from pretalx.event.models import Event, Team
+from pretalx.person.models import User
 
 
-class Command(BakeryBuildCommand):
-    help = 'Exports event schedule as a static HTML dump'
+@contextlib.contextmanager
+def fake_admin(event):
+    with rolledback_transaction():
+        user = User.objects.create_user(
+            email=f'export-{event.slug}@pretalx.invalid',
+            name=f'export-{event.slug}',
+            locale=event.locale,
+        )
+        team = Team.objects.create(
+            organiser=event.organiser,
+            name=f'export-team-{event.slug}',
+            can_change_event_settings=True,
+        )
+        team.limit_events.add(event)
+        team.members.add(user)
 
-    def __init__(self, *args, **kwargs):
-        self._exporting_event = None
-        super().__init__(*args, **kwargs)
+        c = Client()
+        c.force_login(user)
+
+        def get(url):
+            response = c.get(url, is_html_export=True, HTTP_ACCEPT='text/html')
+            content = get_content(response)
+            return content
+
+        yield get
+
+
+def find_assets(html):
+    """ find URLs of additional files needed to render `html`, e.g. images, js, css, ... """
+    soup = BeautifulSoup(html, "lxml")
+
+    for asset in soup.find_all(['script', 'img', 'link']):
+        if asset.name in ['script', 'img']:
+            yield asset.attrs['src']
+        elif asset.name == 'link':
+            if asset.attrs['rel'][0] in ['icon', 'stylesheet']:
+                yield asset.attrs['href']
+        else:
+            raise NotImplementedError()
+
+
+def find_urls(css):
+    return re.findall(r'url\("?(/[^")]+)"?\)', css.decode('utf-8'), re.IGNORECASE)
+
+
+def urls_event_talks(event):
+    talks = event.submissions.filter(
+        pk__in=event.current_schedule.slots.all().values_list('pk', flat=True)
```

Comment:
I'm confused about this line – what's te difference as opposed to `talks = event.current_schedule.slots`?

Reply:
You might also just want to use `event.talks`, come to think of it.